ramips: add support for Mercury KM08-708H

momothefox opened the pull request:
Mercury
MSIP-CMM-MCS-KM08-708H
KM08-708H

SoC : MediaTek MT7621AT
RAM : 256M (Hanya NT5CC128M16IP-DI)
FLASH : NAND (Winbond W29N01HVSINA)
WiFi : MediaTek MT7615DN DUAL Band
BTN : Reset
BTN : WPS
USB: 1 port

LEDS :

WAN Green
LAN {1-4} Green
WiFi Green
UART : J1 TX - RX - 3V3 - GND / 115200-8N1

UART Menu
```
U-Boot 1.1.3 (Apr 14 2017 - 11:33:48)

Board: Ralink APSoC DRAM:  256 MB

 ##### The CPU freq = 880 MHZ ####

Please choose the operation:
   1: Load system code to SDRAM via TFTP.
   2: Load system code then write to Flash via TFTP.
   3: Boot system code via Flash (default).
   4: Entr boot command line interface.
   7: Load Boot Loader code then write to Flash via Serial.
   9: Load Boot Loader code then write to Flash via TFTP.
default: 3
```
Steps

Prepare TFTP server on your Computer

Connect UART Console

Press `4: Entr boot command line interface.`
remove checksum `mcr remove_checksum`
reset
Press `2: Load system code then write to Flash via TFTP.`

Enter IP information and file name

After uploading firmware image, device will boot Openwrt.

Signed-off-by: Muhammad AL-Qady [m.ismael@gmail.com]

hauke commented on target/linux/ramips/dts/mt7621_mercury_km08-708h.dts:
```diff
@@ -0,0 +1,191 @@
+		partition@80000 {
+			label = "u-boot-env";
+			reg = <0x80000 0x40000>;
+
```
remove empty line

momothefox replied:
done.

hauke commented on target/linux/ramips/dts/mt7621_mercury_km08-708h.dts:
```diff
@@ -0,0 +1,191 @@
+	wifi@0,0 {
+		compatible = "mediatek,mt76";
+		reg = <0x0000 0 0 0 0>;
+		mediatek,mtd-eeprom = <&bdinfo 0x0>;
```
Do not use `ediatek,mtd-eeprom`, but:
```
		nvmem-cells = <&eeprom1>;
		nvmem-cell-names = "eeprom";
```

momothefox replied:
done.

hauke commented on target/linux/ramips/mt7621/base-files/etc/board.d/01_leds:
```diff
@@ -150,6 +150,10 @@ linksys,ea8100-v2)
 	;;
+mercury,km08-708h)
+	ucidef_set_led_netdev "lan2" "lan2 link" "green:lan2" "lan2" "link tx rx"
+	ucidef_set_led_netdev "wlan" "wlan link" "green:wlan" "wlan0" "link tx rx"
```
This should not be needed with your `linux,default-trigger = "phy0radio";` definition.

momothefox replied:
still needed. it did not generate led config when i removed it.

hauke replied:
For the LAN port it is needed, for WLAN it should work without a configuration.

momothefox commented on target/linux/ramips/dts/mt7621_mercury_km08-708h.dts:
```diff
@@ -0,0 +1,191 @@
+	};
+};
+
+&gpio {
+	usbpower-hog {
+		gpio-hog;
+		gpios = <15 GPIO_ACTIVE_LOW>;
+		output-high;
+		line-name = "usbpower";
+	};
+};
```
this is from file `/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/image-mt7621_mercury_km08-708h.dtb`
```
	regulator-vbus {
		compatible = "regulator-fixed";
		regulator-max-microvolt = <0x4c4b40>;
		regulator-min-microvolt = <0x4c4b40>;
		regulator-name = "usb_power";
		gpios = <0x01 0x0f 0x01>;
		enable-active-high;
		phandle = <0x0f>;
	};

```
Tested it, did not work

momothefox committed:
ramips: add support for Mercury KM08-708H

Signed-off-by: Muhammad Al-Qadhy <m.ismael@gmail.com>

Mercury
MSIP-CMM-MCS-KM08-708H
KM08-708H

SoC : MediaTek MT7621AT
RAM : 256M (Hanya NT5CC128M16IP-DI)
FLASH : NAND (Winbond W29N01HVSINA)
WiFi : MediaTek MT7615DN DUAL Band
BTN : Reset
BTN : WPS
USB: 1 port

LEDS :

WAN Green
LAN {1-4} Green
WiFi Green
UART : J1 TX - RX - 3V3 - GND / 115200-8N1

UART Menu
U-Boot 1.1.3 (Apr 14 2017 - 11:33:48)

Board: Ralink APSoC DRAM:  256 MB

 ##### The CPU freq = 880 MHZ ####

Please choose the operation:
   1: Load system code to SDRAM via TFTP.
   2: Load system code then write to Flash via TFTP.
   3: Boot system code via Flash (default).
   4: Entr boot command line interface.
   7: Load Boot Loader code then write to Flash via Serial.
   9: Load Boot Loader code then write to Flash via TFTP.
default: 3
Steps
Prepare TFTP server on your Computer
Connect UART Console

Press 4: Entr boot command line interface.
remove checksum mcr remove_checksum
reset
Press 2: Load system code then write to Flash via TFTP.

Enter IP information and file name

After uploading firmware image, device will boot Openwrt.

Signed-off-by: Muhammad Al-Qadhy <m.ismael@gmail.com>

momothefox committed:
ramips: add support for Mercury KM08-708H
Signed-off-by: Muhammad Al-Qadhy <m.ismael@gmail.com>

Mercury
MSIP-CMM-MCS-KM08-708H
KM08-708H

SoC : MediaTek MT7621AT
RAM : 256M (Hanya NT5CC128M16IP-DI)
FLASH : NAND (Winbond W29N01HVSINA)
WiFi : MediaTek MT7615DN DUAL Band
BTN : Reset
BTN : WPS
USB: 1 port

LEDS :

WAN Green
LAN {1-4} Green
WiFi Green
UART : J1 TX - RX - 3V3 - GND / 115200-8N1

UART Menu
U-Boot 1.1.3 (Apr 14 2017 - 11:33:48)

Board: Ralink APSoC DRAM:  256 MB

 ##### The CPU freq = 880 MHZ ####

Please choose the operation:
   1: Load system code to SDRAM via TFTP.
   2: Load system code then write to Flash via TFTP.
   3: Boot system code via Flash (default).
   4: Entr boot command line interface.
   7: Load Boot Loader code then write to Flash via Serial.
   9: Load Boot Loader code then write to Flash via TFTP.
default: 3
Steps
Prepare TFTP server on your Computer
Connect UART Console

Press 4: Entr boot command line interface.
remove checksum mcr remove_checksum
reset
Press 2: Load system code then write to Flash via TFTP.

Enter IP information and file name

After uploading firmware image, device will boot Openwrt.

Signed-off-by: Muhammad Al-Qadhy <m.ismael@gmail.com>

hauke commented on target/linux/ramips/dts/mt7621_mercury_km08-708h.dts:
```diff
@@ -0,0 +1,210 @@
+		compatible = "gpio-leds";
+
+		led_wlan: wlan {
+			label = "green_wlan";
```
The label is not needed any more, it will automatically generate the `green:wlan` label. Same for the other LED.

momothefox replied:
this apply to v24.10.2 ? because i am testing on it.

momothefox commented on target/linux/ramips/dts/mt7621_mercury_km08-708h.dts:
```diff
@@ -0,0 +1,210 @@
+	model = "Mercury KM08-708H";
+
+	aliases {
+		label-mac-device = &wan;
```
@MShokry sorry i got very busy.
i will try to test these and push the code for review

abdoHarb93 commented:
Hello,
There's any chance to finish soon?
